Improved some Widgets

quantenschaum opened the pull request:
- reuse existing formatters in widgets
- speed widget
  - added Beaufort
  - added aliases `m/s` and `km/h`
- direction widget
  - added 180 degree option
- distance widget
  - added feet and yards units (for depth)
- Wind Widgets
  - allow to configure speed
  - 180 degree option
  - red/green graphics for angles
- depth widget
  - added options for below surface, keel, transducer
  - added unit config option
- position and map center widget
  - position format options: DD = decimal degrees, DDM = degrees + decimal minutes, DMS = degrees + minutes + seconds, OLC = open location code
  - position in monospace font for better readability (when line wrapped)
  - map center widget: configurable distance unit
- AIS data
  - consistent field names
  - units after values
  - order and formatting of data in target list
  - use labels and units defined in aisformatter

quantenschaum commented:
Would be nice if you had a look and merge this.

wellenvogel commented:
Thx a lot for all the ideas/changes.
But bundling all of them into one PR makes it really hard to merge. I always have to check the impact and I would really prefer small PRs for each of the proposed changes.

quantenschaum commented:
It's actually easy to merge, it's just one click. :clown_face: 

The changes are somewhat similar and some are connected. It seems a bit tedious to go through them, but it's not too hard and worth it IMHO.

quantenschaum commented:
1. license of open-location-code: Apache 2, does that work with avnav? - OLC is not strictly necessary, but it is a very nice and compact format IMHO
2. `km/s` should be `km/h`, thanks for spotting my mistake. I added these units because like this they are correct `ms != m/s`. Since I change the widgets to display these strings as unit, I want them to be correct. I left the old ones for compatibility.
3. Where should the descriptions go? - Does `km/h` really need a description?
4. `let/var` - OK, but it is not always a problem, potentially yes. I will change this, but there are places where `var` is appropriate.
5. removed `var`, what is obsolete
6. removed `var`, what is obsolete
7. I will try to separate it, but if you merged both, it would not matter.

quantenschaum commented:
7. I looked into it, actually it does contain stuff from the CPA issue, there are just conflicts because ais widget/page was changed here as well.
